ompi/mca/coll/basic/coll_basic_neighbor_allgather.c

```diff
@@ -143,16 +145,20 @@ mca_coll_basic_neighbor_allgather_graph(const void *sbuf, int scount,
          * a const for the send buffer. */
         rc = MCA_PML_CALL(isend((void *) sbuf, scount, sdtype, edges[neighbor],
                                 MCA_COLL_BASE_TAG_ALLGATHER, MCA_PML_BASE_SEND_STANDARD,
-                                comm, reqs++));
+                                comm, preqs++));
         if (OMPI_SUCCESS != rc) break;
     }
 
     if (OMPI_SUCCESS != rc) {
-        /* should probably try to clean up here */
+        ompi_coll_base_free_reqs( reqs, (2 * neighbor + 1));
```

Comment:
@bosilca Replace (2*neighbor+1) by a counter incremented before each request
or
by 2*(neighbor+1).
Freeing (2*neighbor+1), we might miss one request if the isend fails